plugins/contactpressuresensorsarray/src/ContactPressureSensorsArray.cc

```diff
@@ -0,0 +1,193 @@
+        if (m_wrapperName != "analogServer")
+        {
+            yError() << "GazeboYarpContactPressureSensorsArray Plugin Failed: \"device\" should be set to \"analogSensor\" network wrapper.";
+            return;
+        }
+    
+        // Check on Required Parameter for Device driver
+        if (driver_properties.find("device").isNull())
+        {
+            yError() << "GazeboYarpContactPressureSensorsArray Plugin Failed: Missing parameter \"device\"  under [DRIVER] in config";
+            return;
+        }
+        m_sensorDriverName = driver_properties.find("device").asString();
+        if (m_sensorDriverName != "gazebo_contactpressuresensorsarray")
+        {
+            yError() << "GazeboYarpContactPressureSensorsArray Plugin Failed: \"device\" should be set to \"gazebo_contactpressuresensorsarray\".";
+            return;
+        }
+    
+        if (driver_properties.find("linkName").isNull())
+        {
+            yError() << "GazeboYarpContactPressureSensorsArray Plugin Failed: Missing parameter \"linkName\"  under [DRIVER] in config";
+            return;
+        }
+
+        // Adding robot name so that it can be used in the driver
+        yarp::os::Bottle& driver_robot_config = driver_properties.addList();
+        driver_robot_config.addString("robotName");
+        driver_robot_config.addString(m_gazeboRobotName.data());
+   
+        // Check if robot name is added to the pluginParameters
+        if (driver_properties.find("robotName").isNull())
+        {
+            yError() << "GazeboYarpContactPressureSensorsArray Plugin Failed: Sensor name not passed to the driver";
+            return;
+        }
+    
+        // Open the wrapper
+        if (!m_dev_wrapper.open(wrapper_properties))
+        {
+            yError() << "GazeboYarpContactPressureSensorsArray Plugin failed: error in opening wrapper for yarp driver";
+            return;
+        }
+    
+        // Open the driver
+        if (!m_dev_driver.open(driver_properties))
+        {
+            yError() << "GazeboYarpContactPressureSensorsArray Plugin failed: error in opening yarp driver";
+            return;
+        }
+    
+        // Attach the driver to the wrapper
+        yarp::dev::PolyDriverList driver_list;
+    
+        if (!m_dev_wrapper.view(m_imultwrapper))
+        {
+            yError() << "GazeboYarpContactPressureSensorsArray Plugin failed: Error in loading wrapper";
+            return;
+        }
+    
+        driver_list.push(&m_dev_driver, "dummy");
+    
+        if (!m_imultwrapper->attachAll(driver_list))
+        {
+            yError() << "GazeboYarpContactPressureSensorsArray Plugin failed: Error in connecting the wrapper and device";
+        }
+     
+        yInfo() << "GazeboYarpContactPressureSensorsArray: Opened device driver and attached to Analog Server wrapper \n";
+    } // end Load(sensors::SensorPtr _parent, sdf::ElementPtr _sdf)
```

Comment:
Note: I am not a big fan of this "end of scope" comments, as it is difficult to maintain them when you change names of functions. However if you like them we can keep them.

Reply:
I had just put it there, since it was a really big function block... I will remove it!